kata-ctl:  Add the option to install kata-ctl to a user specified directory

ngpatel6 opened the pull request:
Fixes #5403 

The Makefile was updated to use an INSTALL_PATH variable to track where the kata-ctl binary should be installed.  If the user doesn't specify anything, then it uses the default path that cargo uses.  Otherwise, it will install it in the directory that the user specified.  The README.md file was also updated to show how to use the new option.

This work was done by pair programming by the following contributors: Cesar Tamayo
Kevin Mora Jimenez
Narendra Patel
Ray Karrenbauer
Srinath Duraisamy

Fixes #5403

amshinde reviewed (approved):
@ngpatel6 lgtm. You can attribute this commit to all the authors that worked on this by using multiple "Co-authored-by: " lines. 
Take a look at: https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

That way everyone who worked on the change can receive credit for their work. We have also accepted commits that have multiple "Signed-off-by: " lines in the past.

cmaf reviewed (commented):
@ngpatel6 lgtm, pending @amshinde's suggestion to add co-authors

jodh-intel reviewed (approved):
Thanks @ngpatel6.

lgtm

liubin commented on src/tools/kata-ctl/README.md:
````diff
@@ -27,6 +27,11 @@ $ make
 
+If you would like to install the tool to a specific directory, then you can provide it through the `INSTALL_PATH` variable.
+```bash
+$ make install INSTALL_PATH=<install_path>
````
This line cause a static check error: `ERROR: shell code in file 'src/tools/kata-ctl/README.md' is not valid`.
Maybe use a real example will pass:

```
$ make install INSTALL_PATH=/opt/bin
```

jodh-intel commented:
Hi @ngpatel6 - The CI is failing due to the commit format:

```
Error: 

  7c2988e020f488c5556bb08ad3c6204f303305b9    Failed to find subsystem in subject
```

See the document below for help on formatting commits for the project.

https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format

Please can you `git commit --amend` and update the commit format to match the format specified in the doc above. Something like this:

```
kata-ctl: Allow INSTALL_PATH= to be specified

Update the kata-ctl install rule to allow it to be installed to a given directory

The Makefile was updated to use an `INSTALL_PATH` variable to track where the
kata-ctl binary should be installed.  If the user doesn't specify anything,
then it uses the default path that cargo uses.  Otherwise, it will install it
in the directory that the user specified.  The README.md file was also updated
to show how to use the new option.

This work was done by pair programming by the following contributors:
Cesar Tamayo
Kevin Mora Jimenez
Narendra Patel
Ray Karrenbauer
Srinath Duraisamy

Fixes: #5403.

Signed-off-by: Narendra Patel <narendra.g.patel@intel.com>
```

ngpatel6 commented:
Apologies team, been busy with the normal workload after the hackathon.  I'll be addressing these issues this week.  Thanks for all the quick feedback!

amshinde commented:
@ngpatel6 Any updates on this PR? Would like to get this merged soon.

ngpatel6 commented:
Apologies for the late update.  I've made the suggested updates and it should be ready for a final review.

chavafg commented:
/test
